sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -115,14 +128,14 @@ function buildPipelineRequest(
   url: string,
   options: InternalRequestParameters = {}
 ): PipelineRequest {
-  const { body, formData } = getRequestBody(options.body, options.contentType);
+  const { body, formData } = getRequestBody(options.body, getRequestContentType(options));
```

Comment:
after changing it this way, I found we will always fall back to application/json when we build the request body.